routing/index_graph.hpp

```diff
@@ -77,16 +119,154 @@ class IndexGraph final
+    Build(jointsSize);
+  }
+
+  Joint::Id GetJointIdForTesting(RoadPoint const & ftp) const
+  {
+    return m_roadIndex.GetJointId(ftp);
+  }
+
+  /// \brief Disable all edges between |from| and |to| if they are different and adjacent.
+  /// \note Despit the fact that |from| and |to| could be connected with several edges
+  /// it's a rare case. In most cases |from| and |to| are connected with only one edge
+  /// if they are adjacent.
+  /// \note The method doesn't affect routing if |from| and |to| are not adjacent or
+  /// if one of them is equal to Joint::kInvalidId.
+  void DisableEdge(Joint::Id from, Joint::Id to)
+  {
+    m_blockedEdges.insert(DirectedEdge(from, to));
   }
+  /// \brief Adding a fake oneway feature with a loose end starting from joint |from|.
+  /// Geometry for the feature points is taken from |geometrySource|.
+  /// If |geometrySource| contains more than two points the feature is created
+  /// with intermediate (not joint) point(s).
+  /// \returns feature id which was added.
+  uint32_t AddFakeLooseEndFeature(Joint::Id from, vector<RoadPoint> const & geometrySource);
+
+  /// \brief Connects joint |from| and |to| with a fake oneway feature. Geometry for the feature
+  /// points is taken from |geometrySource|. If |geometrySource| contains more than
+  /// two points the feature is created with intermediate (not joint) point(s).
+  /// \returns feature id which was added.
+  uint32_t AddFakeFeature(Joint::Id from, Joint::Id to, vector<RoadPoint> const & geometrySource);
+
+  /// \brief Adds restriction to navigation graph which says that it's prohibited to go from
+  /// |restrictionPoint.m_from| to |restrictionPoint.m_to|.
+  /// \note |from| and |to| could be only begining or ending feature points and they has to belong
+  /// to
+  /// the same junction with |jointId|. That means features |from| and |to| has to be adjacent.
+  /// \note This method could be called only after |m_roadIndex| have been loaded with the help of
+  /// Deserialize() or Import().
+  void ApplyRestrictionNo(RestrictionPoint const & restrictionPoint);
+
+  /// \brief Adds restriction to navigation graph which says that from feature
+  /// |restrictionPoint.m_from| it's permitted only
+  /// to go to feature |restrictionPoint.m_to|. All other ways starting form
+  /// |restrictionPoint.m_form| is prohibited.
+  /// \note All notes which are valid for ApplyRestrictionNo() is valid for ApplyRestrictionOnly().
+  void ApplyRestrictionOnly(RestrictionPoint const & restrictionPoint);
+
+  void ApplyRestrictionOnlyRealFeatures(RestrictionPoint const & restrictionPoint);
+  void ApplyRestrictionNoRealFeatures(RestrictionPoint const & restrictionPoint);
+
+  /// \brief Add restrictions in |restrictions| to |m_ftPointIndex|.
+  void ApplyRestrictions(RestrictionVec const & restrictions);
+
+  /// \brief Fills |singleFeaturePath| with points from point |from| to point |to|
+  /// \note |from| and |to| has to belong the same feature.
```

Comment:
> |from| and |to| should belong to the same feature

Fixed.

> RoadPoint is actually a pair - feature id and segment id. So, if feature id should be the same, why not to pass directly feature id and a couple of segment ids?

Yes. It's a possible solution. But are you sure it'll be better?
Now there are following advantages:
(1) logically this method fills a vector of point (type RoadPoint) starting from |from| (type RoadPoint) and ending with |to| (type RoadPoint). I guess it's quite intuitive interface
(2) 2 param instead of 3. 

And disadvantages:
(1) feature id is copied twice in input params (|from| and |to|);
(2) because of it there's CHECK_EQUAL(from.GetFeatureId(), to.GetFeatureId(), ()); in the method.

I guess an advantage (1) is an important one. What do you think about that?